easybuild/easyconfigs/p/preCICE/preCICE-3.1.1-foss-2023a.eb

```diff
@@ -0,0 +1,65 @@
+easyblock = 'CMakeMake'
+
+name = 'preCICE'
+version = '3.1.1'
+
+homepage = 'https://www.precice.org/'
+description = """preCICE (Precise Code Interaction Coupling Environment)
+is a coupling library for partitioned multi-physics simulations,
+including, but not restricted to fluid-structure interaction and
+conjugate heat transfer simulations. Partitioned means that preCICE
+couples existing programs (solvers) capable of simulating a subpart of
+the complete physics involved in a simulation. This allows for the high
+flexibility that is needed to keep a decent time-to-solution for complex
+multi-physics scenarios."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://github.com/precice/precice/archive']
+sources = ['v%(version)s.tar.gz']
+checksums = ['fe759293942ebc9cb2e6127f356a8c795ab7383c1b074595994ebc92466e478d']
+
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('pkgconf', '1.9.5'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('Boost', '1.82.0'),
+    ('Eigen', '3.4.0'),
+    ('libxml2', '2.11.4'),
+    ('PETSc', '3.20.3'),
+    ('SciPy-bundle', '2023.07'),
+]
+
+separate_build_dir = True
+build_shared_libs = True
+
+# Tell pkg-config to keep flags available in CPATH and others because it prevents CMake from finding PETSc correctly
+preconfigopts = 'PKG_CONFIG_ALLOW_SYSTEM_LIBS=1 PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 '
+# Make sure MPI support is enabled
+configopts = '-DPRECICE_MPICommunication=ON '
+# Make sure PETSc support is enabled
+configopts += '-DPRECICE_PETScMapping=ON '
+# Make sure PythonActions support is enabled
+configopts += "-DPYTHON_INCLUDE_DIR=$EBROOTPYTHON/include/python%(pyshortver)s "
+configopts += "-DPYTHON_LIBRARY=$EBROOTPYTHON/lib/libpython%%(pyshortver)s.%s " % SHLIB_EXT
+configopts += '-DPRECICE_PythonActions=ON '
+# force Eigen include directory which is not found correctly otherwise
+configopts += '-DEIGEN3_INCLUDE_DIR="$EBROOTEIGEN/include"'
+
+# Run basic tests
+# runtest = ' test_base'
```

Comment:
Could you explain why this has been disabled (looks like it's still supported: https://github.com/precice/precice/blob/v3.1.1/cmake/CTestConfig.cmake#L449)?

Reply:
These tests fail with OpenMPI 4.1.5 due yo a bug in the OpenMPI library. The bug is described in https://github.com/open-mpi/ompi/issues/11749. Bug fix for OpenMPI is included in pr #20422. After adding this patch to OpenMPI, the precice tests run smoothly.